r/R/dplyr-funcs-datetime.R

```diff
@@ -372,3 +372,50 @@ binding_format_datetime <- function(x, format = "", tz = "", usetz = FALSE) {
 
   build_expr("strftime", x, options = list(format = format, locale = Sys.getlocale("LC_TIME")))
 }
+
+binding_as_date <- function(x,
+                            format = NULL,
+                            tryFormats = "%Y-%m-%d",
+                            origin = "1970-01-01",
+                            tz = "UTC",
+                            base = TRUE) {
+
+  if (is.null(format) && length(tryFormats) > 1) {
+    abort("`as.Date()` with multiple `tryFormats` is not supported in Arrow")
+  }
+
+  if (call_binding("is.Date", x)) {
+    return(x)
+
+    # cast from POSIXct
+  } else if (call_binding("is.POSIXct", x)) {
+    # base::as.Date() first converts to the desired timezone and then extracts
+    # the date, which is why we need to go through timestamp() first
+    if (base || !is.null(tz)) {
+      x <- build_expr("cast", x, options = cast_options(to_type = timestamp(timezone = tz)))
+    }
```

Comment:
Would it be possible to put this exceptional case in the binding for `as.Date` before then calling `binding_as_date`? I'm not 100% sure it would be cleaner, necessarily, but it would make this helper function much cleaner, and keep the exceptional part of the binding with the binding declaration bit.

Reply:
We run row 395 in 2 cases: either when we call `as.Date()` or when we call `as_date()` and the user passes a value for `tz`. So the row is run both by the `as.Date()` and `as_date()` bindings. It encapsulates the main difference between the 2 functions when it comes to the treatment of `tz` (either passed as argument or attribute of the `POSIXct` object)

Reply:
Aaah so maybe it would be helpful to call that argument something else? Like maybe `use_tz`? That way the intent of it is a bit clearer on first read? (And of course comments if they are helpful...)

Reply:
Done. Changed the name to `use_tz` and added, what I think are, clarifying comments.